src/app/utils/expect-entity-data.ts

```diff
@@ -0,0 +1,80 @@
+import { TestBed } from "@angular/core/testing";
+import { EntitySchemaService } from "../core/entity/schema/entity-schema.service";
+import { Entity } from "../core/entity/entity";
+import { EntityMapperService } from "../core/entity/entity-mapper.service";
+
+/**
+ * Run unit test expect to check that all given entities are indeed in the database.
+ * @param expectedEntities array of expected Entity instances to be checked in the database
+ * @param onlyExpected (Optional) if set to true, ensure that the expected entities are the only entities of that type in the database
+ * @param ignoreId do not compare _id of entities, match only based on properties
+ */
+export async function expectEntitiesToBeInDatabase(
+  expectedEntities: Entity[],
+  ignoreId: boolean = false,
+  onlyExpected: boolean = false
+) {
+  const entityMapperService = TestBed.inject(EntityMapperService);
+  const actualData = await entityMapperService.loadType(
+    expectedEntities[0].getConstructor()
+  );
+
+  await expectEntitiesToMatch(
+    actualData,
+    expectedEntities,
+    ignoreId,
+    onlyExpected
+  );
+}
+
+/**
+ * Run unit test expect to compare the given sets of entities while ignoring technical properties
+ * @param actualEntities array of actual Entity instances to be compared
+ * @param expectedEntities array of expected Entity instances to be compared
+ * @param ignoreId do not compare _id of entities, match only based on properties
+ * @param onlyExpected (Optional) if set to true, ensure that the expected entities are the only entities of that type in the database
+ */
+export async function expectEntitiesToMatch(
+  actualEntities: any[],
+  expectedEntities: Entity[],
+  ignoreId: boolean = false,
+  onlyExpected: boolean = true
+) {
+  if (onlyExpected) {
+    expect(actualEntities.length).toBe(expectedEntities.length);
+  }
+
+  const cleanExpected = comparableEntityData(expectedEntities, ignoreId);
+  const cleanActual = comparableEntityData(actualEntities, ignoreId);
+
+  for (const data of cleanExpected) {
+    const passed = expect(cleanActual).toContain(data);
```

Comment:
I would recommend using [custom jasmine matchers](https://jasmine.github.io/tutorials/custom_matcher) for this. This would make the integration of the custom checks a little easier to understand.

Reply:
I looked into this but there seems to be no clean and easy way to have good typescript support (probably need to patch the jasmine namespace to pass type checks in the spec files). I'd like to postpone this topic for now although I like the suggestion to follow a more standard approach here.